backend/layers/business/exceptions.py

```diff
@@ -101,6 +101,14 @@ class DatasetUpdateException(BusinessException):
     pass
 
 
+class DatasetIsPublishedException(DatasetUpdateException):
+    pass
+
+
+class DatasetIsNotPublishedException(DatasetUpdateException):
```

Comment:
Is there a better way of say Not Published without using not. Maybe `DatasetIsPrivateException`. The Not could get lost when reading the name.

Reply:
what about a `NotPublishedException`? Can be used for both types of entities.

Reply:
I'm trying to avoid adding negative logic if possible.

Reply:
Besides choosing an easily understandable name, we should also be adding a docstring for each exception to make it clear when they should be raised. 🙏